app/src/main/java/org/simple/clinic/home/report/ReportsUiRenderer.kt

```diff
@@ -1,12 +1,28 @@
 package org.simple.clinic.home.report
 
 import org.simple.clinic.mobius.ViewRenderer
+import org.simple.clinic.util.ValueChangedCallback
 
 class ReportsUiRenderer(
     private val ui: ReportsUi
 ) : ViewRenderer<ReportsModel> {
 
+  private val valueChangedCallback = ValueChangedCallback<ReportsModel>()
```

Comment:
Adding a changed callback on the model is generally not a good idea since we only care about a certain property in it has changed or not.

In this case, since it's a nullable string, it might be a little tricky to track in the value changed callback since it does not support null values. I think it's ok to modify the tests in these scenarios since we are adding a lot of additional complexity here.

Reply:
Removed it. We don't have to modify tests as well, I forgot I added a null check before rendering the reports. So we don't have to use `ValueChangedCallback` or modify any tests.

Reply:
This was initially added to handle cases where we were using `.distinctUntilChanged` to prevent the UI from rendering again and there's no standard way to do this in the Mobius style. Not doing it would cause tests to break since something would get rendered twice instead of once.

Since rendering is mostly cheap and using this adds additional complexity, I have been tending towards modifying the tests instead of using this since those tests will eventually get removed when we add new features to the screens.